### DISCUSSION TITLE
Why is the Laplacian operator not positive definite?

### DISCUSSION THREAD

**Discussion by avaruuser** [libigl/libigl#2113] [Q&A] [answered]
Hello everyone,

I want to solve a Poisson equation to obtain the curl-free potential of the Helmholtz decomposition of a vector field defined on a mesh using [sksparse's sparse Cholesky decomposition](https://scikit-sparse.readthedocs.io/en/latest/cholmod.html). The Cholesky decomposition requires the Laplacian to be symmetric positive definite, which I believe it should always be. I understand that the discrete Laplacian returned by igl.cotmatrix() should be negative semi-definite. However, the discrete negative laplacian obtained with igl.cotmatrix() from the cow and bunny meshes are symmetric but not positive definite. What am I getting wrong here?

**Comment by alecjacobson** [comment, marked as answer]
It's negative semi-definite. So, -igl.cotmatrix() will be positive
semi-definite.

**Reply by avaruuser** [reply]
My bad, I hadn't considered a numerical tolerance when checking the eigenvalues. Thanks!
